geom.go

```diff
@@ -193,6 +193,19 @@ func (l Layout) ZIndex() int {
 	}
 }
 
+// TransformInPlace replaces all coordinates in g using f.
+func TransformInPlace(g T, f func(Coord, Layout)) T {
```

Comment:
@icholy Do you think we should also pass in `Layout` here? It's effectively constant for any call to `TransformInPlace`.

Reply:
In (what I assume is) the common case, people will be using a closure with access to the layout if they need it. The only time it would be useful is if you had a package of pre-defined transform functions which needed the layout.

``` go
geom.TransformInPlace(g, proj.Reproject)
```

but then you can just do:

``` go
geom.TransformInPlace(g, func(c geom.Coord) geom.Coord {
    return proj.Reproject(c, g.Layout())
})
```

I don't need it for my application, but I don't have a strong opinion either way.

Reply:
Maybe it's worth a benchmark.

Reply:
Thanks, I agree with you that people can use a closure with access to the layout if they need it, so I'll drop layout from the callback.